setup/kubernetes/openshift.md

```diff
@@ -0,0 +1,167 @@
+
+This guide describes customizations to the OpenShift cluster as well as Coder
+that ensure an optimal user experience.
+
+[OpenShift Container Platform]: https://www.openshift.com/products/container-platform
+[Security Context Constraint]: https://docs.openshift.com/container-platform/4.7/authentication/managing-security-context-constraints.html
+
+## Prerequisites
+
+- An OpenShift cluster with a Project (Kubernetes namespace) for Coder
+- OpenShift command-line tools (`oc` and `kubectl`)
+
+## Modify pod and container security contexts
+
+OpenShift's SCC feature enforces particular settings that applications must
+run with. The default `restricted` SCC requires that applications run as a user
+within a project-specific range (`MustRunAsRange`) and must not define a
+seccomp profile.
+
+You can view the restrictions using `oc describe scc restricted`:
+
+```console
+$ oc describe scc restricted
+Name:                                           restricted
+Priority:                                       <none>
+Access:
+  Users:                                        <none>
+  Groups:                                       system:authenticated
+Settings:
+  Allow Privileged:                             false
+  Allow Privilege Escalation:                   true
+  Default Add Capabilities:                     <none>
+  Required Drop Capabilities:                   KILL,MKNOD,SETUID,SETGID
+  Allowed Capabilities:                         <none>
+  Allowed Seccomp Profiles:                     <none>
+  Allowed Volume Types:                         configMap,downwardAPI,emptyDir,persistentVolumeClaim,projected,secret
+  Allowed Flexvolumes:                          <all>
+  Allowed Unsafe Sysctls:                       <none>
+  Forbidden Sysctls:                            <none>
+  Allow Host Network:                           false
+  Allow Host Ports:                             false
+  Allow Host PID:                               false
+  Allow Host IPC:                               false
+  Read Only Root Filesystem:                    false
+  Run As User Strategy: MustRunAsRange
+    UID:                                        <none>
+    UID Range Min:                              <none>
+    UID Range Max:                              <none>
+  SELinux Context Strategy: MustRunAs
+    User:                                       <none>
+    Role:                                       <none>
+    Type:                                       <none>
+    Level:                                      <none>
+  FSGroup Strategy: MustRunAs
+    Ranges:                                     <none>
+  Supplemental Groups Strategy: RunAsAny
+    Ranges:                                     <none>
+```
+
+You can override the default settings by defining the following in your Helm
+Values file:
+
+```yaml
+coderd:
+  podSecurityContext:
+    runAsUser: null
+    seccompProfile: null
+  securityContext:
+    seccompProfile: null
+```
```

Comment:
Ah we will make them disable the runAsUser. Do we need to also null out the `runAsGroup`?

Reply:
Hmm, I didn't change runAsGroup, and deploying to our own OpenShift cluster seemed to work fine:

```
  Supplemental Groups Strategy: RunAsAny
    Ranges:				<none>
```

Reply:
I guess that answers that then